internal/logstreamer/logstreamer.go

```diff
@@ -0,0 +1,129 @@
+package logstreamer
+
+import (
+	"context"
+	"io"
+	"log"
+	"sync"
+	"time"
+)
+
+const defaultBufSize = 1024
+
+type LogStreamer struct {
+	reader  io.ReadCloser
+	broker  *Broker
+	buf     *SyncBuf
+	mu      sync.Mutex
+	streams []chan []byte
+}
+
+func NewLogStreamer(reader io.ReadCloser) *LogStreamer {
+	broker := NewBroker()
+	sub := broker.Subscribe()
+	buf := NewSyncBuf(sub)
+
+	ls := &LogStreamer{
+		reader:  reader,
+		broker:  broker,
+		buf:     buf,
+		streams: []chan []byte{},
+	}
+
+	go ls.readLogs()
+	go broker.Broadcast()
+
+	return ls
+}
+
+func (s *LogStreamer) readLogs() {
+	for {
+		pack := make([]byte, defaultBufSize)
+		_, err := s.reader.Read(pack)
+		if err == io.EOF {
+			// There is nothing left to read now, but the task is still
+			// running - so just chill for a second and try again
+			time.Sleep(time.Second)
```

Comment:
I agree about using context instead of sleeping for cgroup package, will fix that.
But I think I need to clarify why do you think this is a bad idea to use it in the logstreamer goroutine?

The main reason I decided to use sleep was to make the following scenario work:
When the user `run teleworker stream <uuid>`  we will give him all the logs that the job produced until that moment, but also will keep him updated in real-time until either a) the job is finished/terminated or b) the user canceled the stream himself (ctrl+c).
In order to achieve that we should not consider io.EOF as the reason to stop the stream, instead of that it makes sense to wait if anything else can be read from the `reader`

Reply:
Or do you simply mean using something similar to https://github.com/spirifoxy/teleworker/pull/2/files#diff-e069f94d8456139a17bfa925387104858808e2afada3daeee2a510dca15e9d08R103 without changing the logic itself, just replacing sleep with ticker?